rust/lance-encoding/src/encodings/physical/binary.rs

```diff
@@ -739,13 +739,92 @@ impl MiniBlockDecompressor for BinaryMiniBlockDecompressor {
     }
 }
 
+#[derive(Debug, Default)]
+pub struct BinaryBlockEncoder {}
+impl BlockCompressor for BinaryBlockEncoder {
+    fn compress(&self, data: DataBlock) -> Result<LanceBuffer> {
+        let num_values: u32 = data
+            .num_values()
+            .try_into()
+            .expect("The Maximum number of values BinaryBlockEncoder can work with is u32::MAX");
+
+        match data {
+            DataBlock::VariableWidth(mut variable_width_data) => {
+                if variable_width_data.bits_per_offset != 32 {
+                    panic!("BinaryBlockEncoder only works with 32 bits per offset VariableWidth DataBlock.");
+                }
+                let offsets = variable_width_data.offsets.borrow_to_typed_slice::<u32>();
+                let offsets = offsets.as_ref();
+                // the first 4 bytes store the number of values, then 4 bytes for bytes_start_offset,
+                // then offsets data, then bytes data.
+                let bytes_start_offset = std::mem::size_of_val(offsets) as u32 + 4 + 4;
```

Comment:
Since you're using `u32` directly above it seems you may as well do `std::mem::size_of::<u32>()` instead of `size_of_val`.

Reply:
I am using the `size_of_val` to get the whole length in bytes of &[u32] here,
`std::mem::size_of::<u32>()` can only get me the size of a element.